cluster/engine.go

```diff
@@ -53,6 +55,14 @@ type Engine struct {
 	eventHandler    EventHandler
 	healthy         bool
 	overcommitRatio int64
+	ScheduleQueue   *list.List
+}
+
+//ScheduledItem FIX ME : works for affinities as it needs the conatainer name and Image
+//Need to add more types to support all filters
+type ScheduledItem struct {
```

Comment:
```scheduler/node``` has the list of containers that are already created but not the list of containers that are scheduled but not created.  . I think if we try to inject or add container before its creation it hits back during refreshloop while updating list of containers, as we don't have complete details to make a container struct or Image struct until the container gets created